packages/block-library/src/social-links/style.scss

```diff
@@ -16,19 +16,56 @@
 		border-bottom: 0;
 		box-shadow: none;
 	}
+
+	// Vertically balance the margin of each icon.
+	.wp-social-link {
+		margin: 4px 8px 4px 0;
+
+		// By setting the font size, we can scale icons and paddings consistently based on that.
+		// This also allows themes to override this, if need be.
+		a {
+			padding: 0.25em;
+		}
+
+		svg {
+			width: 1em;
+			height: 1em;
+		}
+	}
+
+	// Icon sizes.
+	// Small.
+	&.has-small-icon-size {
+		font-size: 16px; // 16 makes for a quarter-padding that keeps the icon centered.
+	}
+
+	// Normal/default.
+	&,
+	&.has-normal-icon-size {
+		font-size: 24px;
+	}
+
+	// Large.
+	&.has-large-icon-size {
+		font-size: 36px;
+	}
+
+	// Huge.
+	&.has-huge-icon-size {
+		font-size: 48px;
```

Comment:
for the font-sizes you could do with size variables, since they are all multiples of 8

Reply:
I tend to avoid variables unless they are to be reused in a number of places, and especially in the style.scss files, as they are just meant for the frontend.